src/config/ConfigManager.cpp

```diff
@@ -873,11 +873,15 @@ CConfigManager::CConfigManager() {
 std::optional<std::string> CConfigManager::generateConfig(std::string configPath) {
     std::string parentPath = std::filesystem::path(configPath).parent_path();
 
-    if (!std::filesystem::is_directory(parentPath)) {
-        Debug::log(WARN, "Creating config home directory");
-        try {
-            std::filesystem::create_directories(parentPath);
-        } catch (std::exception& e) { throw e; }
+    if (!parentPath.empty()) {
+        std::error_code ec;
+        bool            created = std::filesystem::create_directories(parentPath, ec);
+        if (ec) {
```

Comment:
can this set ec if already exists? Already existing is not fatal

Reply:
If the config file already exists then `created` will be `false`. `if (ec)` will evaluate to false. I tested this behavior by compiling this on to a fresh install (in a vm) of Arch. I was able to run Hyprland, and it created the example config file. I then exited out of Hyprland and then ran it again. I did not receive any errors.